solidity/src/BftBridge.sol

```diff
@@ -181,8 +181,10 @@ contract BFTBridge is TokenManager, UUPSUpgradeable, OwnableUpgradeable, Pausabl
             order.feePayer != address(0) && msg.sender == minterCanisterAddress
                 && address(feeChargeContract) != address(0)
         ) {
-            uint256 gasFee = initGasLeft - gasleft() + additionalGasFee;
-            uint256 fee = gasFee * tx.gasprice;
+            uint256 additionalGasConsumed = initGasLeft - gasleft();
+            uint256 additionalGasFeeConsumed = (additionalGasConsumed * 115) / 100; // Add 15% buffer
+            uint256 fee = additionalGasFeeConsumed * tx.gasprice;
```

Comment:
can you add a test to check if the fee is the expected one?

Reply:
The code has been re-worked